### PR TITLE
Adds captcha to UGP flow

### LINKED ISSUE #14188: UGP captcha
### Steps to Reproduce

  1.
  2.
  3.

**Actual result:**

**Expected result:**

**Reproduces how often:**

### Brave Version

**about:brave info:**

**Reproducible on current live release:**

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves #14188 

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
- clean profile
- start browser with staging flag
- enable payments
- claim promotion
- verify that you can see captcha overlay
- solve (and try to miss) the captcha
- make sure that success overlay is displayed
- close overlay and make sure that you received 10BAT

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: test/unit/app/browser/reducers/ledgerReducerTest.js [hunk 1]
```diff
@@ -969,4 +971,12 @@ describe('ledgerReducer unit tests', function () {
...
     })
   })
+
+  describe('APP_ON_PROMOTION_CLICK', function () {
+
+  })
+
+  describe('APP_ON_CAPTCHA_RESPONSE', function () {
+
+  })
```
Comment (jasonrsadler) [inline test/unit/app/browser/reducers/ledgerReducerTest.js]: Do we need tests on these two?

Comment (NejcZdovc) [reply test/unit/app/browser/reducers/ledgerReducerTest.js]: lol I added them but then forgot to actually add them 😄 thx

Comment (NejcZdovc) [reply test/unit/app/browser/reducers/ledgerReducerTest.js]: tests added

File: app/renderer/components/preferences/payment/captcha.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+const ImmutableComponent = require('../../immutableComponent')
+
+// Actions
+const appActions = require('../../../../../js/actions/appActions')
+
+// Constants
+const promotionStatuses = require('../../../../common/constants/promotionStatuses')
+
+// Styles
+const cx = require('../../../../../js/lib/classSet')
```
Comment (jasonrsadler) [inline app/renderer/components/preferences/payment/captcha.js]: Were we deprecating the use of className cx()? It's implied from the styling guidelines but I don't know for sure.

Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/captcha.js]: not sure

File: app/renderer/components/preferences/payment/enabledContent.js [hunk 3]
```diff
@@ -632,6 +644,23 @@ const styles = StyleSheet.create({
     }
   },
 
+  enabledContent__captcha: {
+    // TODO we need to make sure that text is not in DND zone
```
Comment (jasonrsadler) [inline app/renderer/components/preferences/payment/enabledContent.js]: These don't look like they're used here. They're defined in the captcha component as well.

File: app/renderer/components/preferences/payment/captcha.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+  render () {
+    const text = this.getText()
+
+    return <div
+      className={cx({
+        [css(styles.enabledContent__overlay)]: true,
+        [css(styles.enabledContent__captcha)]: true,
+        'enabledContent__overlay': true
+      })}
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/captcha.js]: done

File: app/renderer/components/preferences/payment/enabledContent.js (hunk 3 above)
Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/enabledContent.js]: fixed

File: app/renderer/components/preferences/payment/captcha.js [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    }
+  }
+
+  render () {
+    const text = this.getText()
+
+    return <div
+      className={cx({
+        [css(styles.enabledContent__overlay, styles.enabledContent__captcha)]: true,
+        'enabledContent__overlay': true
```
Comment (jasonrsadler) [inline app/renderer/components/preferences/payment/captcha.js]: I meant:
is `styles.enabledContent__overlay` and the `enabledContent__overlay` below it reference the same styling? If so, then just:
`className={css(styles.enabledContent__overlay, styles.enabledContent__captcha)}`

File: app/renderer/components/preferences/payment/captcha.js (hunk 2 above)
Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/captcha.js]: removed

File: app/renderer/components/preferences/payment/captcha.js (hunk 5 above)
Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/captcha.js]: fixed

**Comment by jasonrsadler** [comment]
Really awesome.

There were a couple times where I had dragged over the outline but was still told to 'try again'. Should we implement an onhover over the outline coordinates [and maybe bolden the outline on hover] to let the user know when they are dropping on the right answer? (I'm assuming the coordinate answer is sent to the client)

**Comment by NejcZdovc** [comment]
@jasonrsadler we don't know the coordinates, we just send dropped coordinates to the server and server then decides if it's ok or not

**Comment by jasonrsadler** [comment]
Had a feeling that's how it was working. Just checking.

**Comment by NejcZdovc** [comment]
master 8e9be72
0.23 97c2bb962
0.22 d99e3909d

**Pull request merged**